core/src/main/java/nl/nn/adapterframework/task/TimeoutGuard.java

```diff
@@ -48,7 +49,13 @@ public Killer() {
 
 		@Override
 		public void run() {
-			log.warn("Thread ["+thread.getName()+"] executing task ["+description+"] exceeds timeout of ["+timeout+"] s, interuppting");
+			if (log.isWarnEnabled()) {
+				String stackTrace = Arrays.stream(thread.getStackTrace())
+						.map(StackTraceElement::toString)
+						.reduce("\n", (acc, element) -> acc + "    at " + element + "\n");
+				log.warn("Thread [{}] executing task [{}] exceeds timeout of [{}s], interrupting. Execution stacktrace:{}",
+						thread.getName(), description, timeout, stackTrace);
```

Comment:
I suggest to only print the stacktrace if the description starts with 'stopping receiver'

Reply:
> I suggest to only print the stacktrace if the description starts with 'stopping receiver'

Or perhaps using a configuration-property in `AppConstants`, to make it less specific in case other issues would need to be debugged in future?

Reply:
Or you only log the stacktrace when isDebugEnabled ;)